drupal/web/modules/custom/wunder_search/src/Plugin/Normalizer/NodeNormalizer.php

```diff
@@ -0,0 +1,276 @@
+   *   The rendered content as a string.
+   */
+  protected function renderEntity(ContentEntityBase $entity, $view_mode, array $context) {
+    if (PHP_SAPI === 'cli') {
+      $this->accountSwitcher->switchTo(new IndexerUserSession());
+    }
+
+    $render_markup = '';
+    $current_active_theme = $this->themeManager->getActiveTheme();
+    // Load the theme object for the theme.
+    $frontend_theme = $this->themeInitialization->initTheme($this->configFactory->get('system.theme')->get('default'));
+    // Switch the theme. This is needed as a renderer might be called from
+    // hook_entity_update in a context with the backend theme active.
+    // See MailsystemManager::mail() for a core example.
+    $this->themeManager->setActiveTheme($frontend_theme);
+    // Render the entity.
+    try {
+      // Get the object language.
+      $langcode = $context['language'] ?? $entity->language()->getId();
+
+      // Build a render array and render it into markup.
+      $view_builder = $this->entityTypeManager
+        ->getViewBuilder($entity->getEntityTypeId());
+      $render_array = $view_builder->view($entity, $view_mode, $langcode);
+
+      $render_markup = $this->renderer->renderPlain($render_array);
+    }
+    finally {
+      // Revert the active theme, this is done inside a finally block so it is
+      // executed even if an exception is thrown during rendering the entity.
+      $this->themeManager->setActiveTheme($current_active_theme);
+
+      // Revert the user session.
+      if (PHP_SAPI === 'cli') {
+        $this->accountSwitcher->switchBack();
+      }
+    }
+    return $render_markup;
+  }
+
+  /**
+   * Normalize entity reference to labels.
+   *
+   * @param \Drupal\Core\Field\EntityReferenceFieldItemList $items
+   *   The entity referenced terms.
+   * @param string $fieldName
+   *   (Optional) Pass the field name from which to extract the label.
+   *
+   * @return array
+   *   An array of labels.
+   */
+  public function normalizeEntityReference(EntityReferenceFieldItemList $items, $fieldName = NULL) {
+    $labels = [];
+    $entitys = $items->referencedEntities();
+
+    foreach ($entitys as $entity) {
+      // Use the custom label field if it exists and has a value.
+      // Otherwise fallback to the entity label.
+      if ($fieldName && $entity->hasField($fieldName) && !empty($entity->get($fieldName)->value)) {
+        $labels[] = $entity->get($fieldName)->value;
+      }
+      else {
+        $labels[] = $entity->label();
+      }
+    }
+
+    return $labels;
+  }
+
+  /**
+   * Transforms the bundle of an entity into a translated string.
+   *
+   * @param \Drupal\Core\Entity\ContentEntityBase $entity
+   *   The entity.
+   * @param array $context
+   *   The context (used to get the language).
+   *
+   * @return string
+   *   The string to use as content label.
+   */
+  protected function getContentTypeLabel(ContentEntityBase $entity, array $context) {
+    $bundle_labels_map = [
```

Comment:
Something like https://drupal.stackexchange.com/a/187983

Reply:
yep, that works.

This comes from a project where they wanted to customize the name of the content type in search results (but not in the admin interface) and translate it into various languages... maybe the simpler version is enough for this template.... I will simplify it.


(unfortunately we cannot add translated bundle names via recipe, because translated configuration is not supported)

Reply:
(changed in https://github.com/wunderio/next4drupal-project/pull/31/commits/f9778462e30aabadcf9b2cfb341a976465cbfaa6 )